bin/improver-wet-bulb-temperature

```diff
@@ -0,0 +1,85 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to generate wet bulb temperatures from air temperature, relative
+   humidity, and pressure data. """
+
+import argparse
+
+import iris
+
+from improver.psychrometric_calculations.psychrometric_calculations import (
+    WetBulbTemperature)
+
+
+def main():
+    """Parser to accept input data and an output destination before invoking
+    the wet bulb temperature plugin. Also accepted is an optional
+    convergence_condition argument that can be used to specify the tolerance of
+    the Newton iterator used to calculate the wet bulb temperatures."""
+
+    parser = argparse.ArgumentParser(
+        description='Calculate a cube of wet bulb temperatures.')
+    parser.add_argument('temperature', metavar='TEMPERATURE',
+                        help='File path to a cube of air temperatures at the '
+                        'points for which the wet bulb temperatures are being '
+                        'calculated.')
+    parser.add_argument('relative_humidity', metavar='RELATIVE_HUMIDITY',
+                        help='File path to a cube of relative humidities at '
+                        'the points for for which the wet bulb temperatures '
+                        'are being calculated.')
+    parser.add_argument('pressure', metavar='PRESSURE',
+                        help='File path to a cube of air pressures at the '
+                        'points for for which the wet bulb temperatures are '
+                        'being calculated.')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
+                        help='The output path for the processed NetCDF.')
+    parser.add_argument('--convergence_condition',
+                        metavar='CONVERGENCE_CONDITION', type=float,
+                        default=0.05,
```

Comment:
I think I am hitting memory issues when I run this on multilevel enukx data on the desktop. We may need some looping over height or similar. I am able to produce a file when the input only has one height level but still has an ensemble dimension.

Reply:
I've added a process() function to the plugin as a wrapper to calculate_wet_bulb_temperature. This slices along the available vertical levels, processes each slice and then merges. I have extended the unit tests to cover this and included a check that all the data is on the same vertical level type (height or pressure, though pressure on pressure levels would be odd); I've not checked the level values (5,10,20 etc) as we are using standardised data.

Reply:
Yes, if we are using pressure levels then we will have to do something slightly different, but this will be covered by the follow on ticket if needed.